corehq/form_processor/models.py

```diff
@@ -1658,6 +1658,9 @@ class Meta(object):
         app_label = "form_processor"
         db_table = LedgerValue_DB_TABLE
         unique_together = ("case", "section_id", "entry_id")
+        index_together = [
+            ("domain", "section_id", "entry_id")
```

Comment:
On ICDS the index for `case, section_id, entry_id` is ~20 GB for the entire cluster (~40 GB including standby copies). I expect that this index will be smaller (because domain names are generally shorter than case_ids).

For comparison the LedgerES index is currently ~50 GB (~100 GB with the replica).

This is only used for generating columns when editing or creating exports. I'd be interested to hear if anyone knows of a more efficient index that we can use for this purpose. This seemed like the most effective solution to me as it allows an index-only lookup.

Both of these solutions seem extremely large for what they're providing. Another solution I considered was to create a new table of these combinations, when a ledger is saved, it should ensure its combination is inserted into that table. And that table would be orders of magnitude smaller than this index. That seemed a little complicated to do at the moment.

Reply:
Not a huge fan of this for quite a niche use case. I think 100% of the time the `entry_id` will always be a product ID (unless there's a bug in the app or you're doing something weird like eNikshay was). There are only a handful of domains using ledgers (10 on prod) so this feels like a lot of extra. On ICDS there is only one section.

What about making it part of the CommTrackConfig object? We could pre-populate it from current data and then require user's to add additional sections manually - or else parse it out of apps.

I realise that may take more effort to get done - so maybe let's just go with this for now. One thought on how to handle this for ICDS would be to remove the index and create a custom function for getting the data which we can set in `localsettings.py`:
```python
LEDGER_EXPORT_COLUMN_GETTER = '.....'
```

Maybe too messy? Also thinking about the move to a more 'custom' world for ICDS and what mechanisms we'd use to do that.